codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/AwsServiceModel.java

```diff
@@ -381,14 +383,28 @@ private MethodSpec getValueForField() {
             result.add(enumMemberGetter(member));
         }
 
+        ClassName autoConstructClass = getAutoConstructClassIfExists(member);
+        if (autoConstructClass != null) {
+            result.add(existenceCheckGetter(member, autoConstructClass));
+        }
+
         result.add(memberGetter(member));
 
         return result.stream();
     }
 
+    private ClassName getAutoConstructClassIfExists(MemberModel member) {
```

Comment:
Nit: `Optional<ClassName>`?

Reply:
Will do; I wasn't aware of the convention you were using.

Reply:
Fixed in 1d86076